images/fcos-bastion-image/root/usr/lib/systemd/system/configure-podman-v6-network.service

```diff
@@ -0,0 +1,16 @@
+[Unit]
+Description=Create podman V6 network
+Wants=network-online.target
+After=network-online.target
+
+[Service]
+ExecStart=/usr/bin/podman network create --ignore --subnet fd1e:7ae1:9e3a:8774::/64 podmanv6
```

Comment:
Let's also touch a file after this command and add a Condition in the unit section to avoid executing this at each boot.

The file can be like `/var/.podman-ipv6-net.done`

Reply:
Let's set ExecStartPre for what comes before the touch command as we've done in the others unless there are specific reasons.

I also wonder if we can set up the subnet from an environment variable and use EnvironmentFile to feed it here.

Reply:
yes we can. let me test and push it

Reply:
wait a sec jad, based on the previous conversation on the jira, I just tried to create a file in the /etc/containers/networks/ directory (podman.json), with this content in a Fedora CoreOS live box:

```json

{
     "name": "podman",
     "id": "41bf9b464d72c21becc8db522b6ca2a5b5d32e9a439b9d9c6c18887bd0d02c94",
     "driver": "bridge",
     "network_interface": "podman0",
     "created": "2024-01-29T17:07:42.470547132Z",
     "subnets": [
          {
               "subnet": "fd1e:7ae1:9e3a:8773::/64",
               "gateway": "fd1e:7ae1:9e3a:8773::1"
          },
          {
               "subnet": "10.88.0.0/16",
               "gateway": "10.88.0.1"
          }
     ],
     "ipv6_enabled": true,
     "internal": false,
     "dns_enabled": false,
     "ipam_options": {
          "driver": "host-local"
     }
}


```

And then I restarted podman by `systemctl restart podman.service` and:

```
root@localhost:~# ip a s
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host noprefixroute 
       valid_lft forever preferred_lft forever
2: enp1s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 52:54:00:f6:20:21 brd ff:ff:ff:ff:ff:ff
    inet 192.168.122.38/24 brd 192.168.122.255 scope global dynamic noprefixroute enp1s0
       valid_lft 3415sec preferred_lft 3415sec
    inet6 fe80::8e2:8f0b:5938:3c38/64 scope link noprefixroute 
       valid_lft forever preferred_lft forever
root@localhost:~# podman run -it -d quay.io/openshifttest/hello-openshift:1.2.0
Trying to pull quay.io/openshifttest/hello-openshift:1.2.0...
Getting image source signatures
Copying blob c0f19c621728 done   | 
Copying blob a0d0a0d46f8b done   | 
Copying blob ac34708d38ec done   | 
Copying blob af51708ecc94 done   | 
Copying blob bcb630a3fb42 done   | 
Copying blob bebfa9e8d9ad done   | 
Copying config 9df0b4cf92 done   | 
Writing manifest to image destination
45a7df1e20d27ebee6db7decf2be0635bcf4a25a381e849044e1078fb70f5117
root@localhost:~# ip a s
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host noprefixroute 
       valid_lft forever preferred_lft forever
2: enp1s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 52:54:00:f6:20:21 brd ff:ff:ff:ff:ff:ff
    inet 192.168.122.38/24 brd 192.168.122.255 scope global dynamic noprefixroute enp1s0
       valid_lft 3395sec preferred_lft 3395sec
    inet6 fe80::8e2:8f0b:5938:3c38/64 scope link noprefixroute 
       valid_lft forever preferred_lft forever
3: podman0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 66:e9:77:ce:32:74 brd ff:ff:ff:ff:ff:ff
    inet 10.88.0.1/16 brd 10.88.255.255 scope global podman0
       valid_lft forever preferred_lft forever
    inet6 fd1e:7ae1:9e3a:8773::1/64 scope global 
       valid_lft forever preferred_lft forever
    inet6 fe80::64e9:77ff:fece:3274/64 scope link proto kernel_ll 
       valid_lft forever preferred_lft forever
4: veth0@if2: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master podman0 state UP group default qlen 1000
    link/ether a6:df:cf:e7:c3:1e brd ff:ff:ff:ff:ff:ff link-netns netns-7536de6c-f6f8-c5e8-f67f-b11ab20ff997
    inet6 fe80::c8ba:1fff:fe19:1703/64 scope link proto kernel_ll 
       valid_lft forever preferred_lft forever
root@localhost:~# podman ps
CONTAINER ID  IMAGE                                        COMMAND           CREATED         STATUS         PORTS       NAMES
45a7df1e20d2  quay.io/openshifttest/hello-openshift:1.2.0  /hello_openshift  13 seconds ago  Up 14 seconds              kind_gauss
root@localhost:~# podman exec -it kind_gauss
Error: must provide a non-empty command to start an exec session: invalid argument
root@localhost:~# podman exec -it kind_gauss bash
bash-5.1# ip a s
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: eth0@if4: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether da:d5:34:cb:6d:3e brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 10.88.0.2/16 brd 10.88.255.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fd1e:7ae1:9e3a:8773::2/64 scope global 
       valid_lft forever preferred_lft forever
    inet6 fe80::d8d5:34ff:fecb:6d3e/64 scope link 
       valid_lft forever preferred_lft forever
bash-5.1# 
```


Which seem a fully enabled dual stack container at the end. Why are we still pursuing the workaround of adding a configuration unit file here? Did you have any issues in your previous testing about just dropping a file to configure the podman0 network?

Reply:
yes, I tested it and it did not work for me. let's revert back to this solution then
I will double check again

Reply:
Try to do it on a FCOS live first and then on an installed system. That's strange as for me it was straightforward with no issues. What didn't work for you?

Reply:
it worked for now. not sure what or how but I missed something:
bash-5.1# ip a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: eth0@if18: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 6e:3c:52:82:00:06 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 10.88.0.6/16 brd 10.88.255.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fd1e:7ae1:9e3a:8773::2/64 scope global
       valid_lft forever preferred_lft forever
    inet6 fe80::6c3c:52ff:fe82:6/64 scope link
       valid_lft forever preferred_lft forever

Reply:
After adding podman.json under: images/fcos-bastion-image/root/etc/containers/networks/, bind9 service comes up successfully and w/o errors